### ISSUE TITLE
Meson, Cmake and PROTOBUF_GENERATE_CPP uses wrong executable

### ISSUE THREAD

**Issue by mdegans** [mesonbuild/meson#7509]
**Describe the bug**
When running PROTOBUF_GENERATE_CPP in a cmake subproject, `protobuf::protoc` is attempted to be used as an executable rather than `protoc`.

**To Reproduce**

1. Clone [this repo](https://github.com/mdegans/libdistance/tree/a974132399ad4f69f6bbe32a85db44e4c848fb2f)
2. Comment out subdirs (since those require proprietary software)
3. attempt to build the "distanceproto" wrap subproject. Output on my system:
```
[4/11] Generating cm_custom_tgt_0 with a custom command
FAILED: subprojects/distanceproto/distance.pb.cc subprojects/distanceproto/distance.pb.h 
/home/mdegans/.local/bin/meson --internal cmake_run_ctgt -o subprojects/distanceproto/distance.pb.cc subprojects/distanceproto/distance.pb.h -O /home/mdegans/Projects/libdistance/builddir/subprojects/distanceproto/__CMake_build/src/distance.pb.cc /home/mdegans/Projects/libdistance/builddir/subprojects/distanceproto/__CMake_build/src/distance.pb.h -d /home/mdegans/Projects/libdistance/builddir/subprojects/distanceproto/__CMake_build/src protobuf::protoc --cpp_out=/home/mdegans/Projects/libdistance/builddir/subprojects/distanceproto/__CMake_build/src -I '/home/mdegans/Projects/libdistance/subprojects/distanceproto/src /home/mdegans/Projects/libdistance/subprojects/distanceproto/src/distance.proto' ';;;'
Traceback (most recent call last):
  File "/home/mdegans/.local/bin/meson", line 8, in <module>
    sys.exit(main())
  File "/home/mdegans/.local/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 228, in main
    return run(sys.argv[1:], launcher)
  File "/home/mdegans/.local/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 217, in run
    return run_script_command(args[1], args[2:])
  File "/home/mdegans/.local/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 165, in run_script_command
    return module.run(script_args)
  File "/home/mdegans/.local/lib/python3.6/site-packages/mesonbuild/scripts/cmake_run_ctgt.py", line 67, in run
    res = subprocess.run(cmd, stdout=stdout, stderr=stderr, cwd=args.directory, check=True)
  File "/usr/lib/python3.6/subprocess.py", line 423, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'protobuf::protoc': 'protobuf::protoc'
[7/11] Compiling C++ object src/libdistance.so.p/FileMetaBroker.cpp.o
ninja: build stopped: subcommand failed.
```

**Expected behavior**
For "protoc" to be called instead of "protobuf::protoc"

**system parameters**
* plain native build.
* Ubuntu 18.04
* Python 3.6.9
* 0.55.0
* 1.8.2

Alternatively (if this is a CMake bug), if the output of a `generator.process()` supported indexing and such generated headers could be installed with `install_headers` I could stop using CMake on the protobuf subproject. Basically I'm trying to isntall generated headers and am not sure if that's possible currently with Meson, so I'm stuck with CMake :/

**Comment by mdegans** [comment]
Related solution for those who might consider switching their subproject to meson as a solution: #7511

**Comment by mensinda** [comment]
Should be fixed with #7527
